File: backend/dto/src/main/java/pl/dawid0604/pcForum/dto/thread/ThreadDTO.java
package pl.dawid0604.pcForum.dto.thread;

import pl.dawid0604.pcForum.dto.post.ThreadNewestPostDTO;

public record ThreadDTO(String encryptedId, String title, String authorEncryptedId,
                        String authorNickname, String authorAvatar, String lastActivityDate,
                        String categoryName, ThreadNewestPostDTO newestPost) { }

File: frontend/src/app/model/ThreadDTO.ts
import { ThreadNewestPostDTO } from "./ThreadNewestPostDTO";


export interface ThreadDTO {
    encryptedId: string,
    title: string,
    authorNickname: string,
    authorEncryptedId: string,
    authorAvatar: string,
    lastActivityDate: string,
    categoryName: string,
    newestPost: ThreadNewestPostDTO,
    numberOfViews: number,
    numberOfPosts: number
}